src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1201,20 +1242,22 @@ bool PolicyHandler::ReceiveMessageFromSDK(const std::string& file,
         correlation_id, vehicle_data_args, application_manager_);
   } else {
     SDL_LOG_WARN("Exchange wasn't successful");
-    policy_manager_->OnPTUFinished(load_pt_result);
+    policy_manager->OnPTUFinished(load_pt_result);
   }
 
   return is_ptu_successful;
 }
 
 bool PolicyHandler::UnloadPolicyLibrary() {
   SDL_LOG_AUTO_TRACE();
-  SDL_LOG_DEBUG("policy_manager_ = " << policy_manager_);
+  auto policy_manager = LoadPolicyManager();
+  DCHECK_OR_RETURN(policy_manager, false);
+  SDL_LOG_DEBUG("policy_manager_ = " << policy_manager);
   bool ret = true;
   AsyncRunner::Stop();
   sync_primitives::AutoWriteLock lock(policy_manager_lock_);
-  if (policy_manager_) {
-    policy_manager_.reset();
+  if (policy_manager) {
+    policy_manager.reset();
```

Comment:
@IGapchuk 
Maybe ExchangePolicyManager(nullptr);

Reply:
@SNiukalov Please, take a look at the new changes in the `fixup` commits. The branch has was force pushed.